AxisMovementConstraint.cs

```diff
@@ -0,0 +1,156 @@
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.SpatialManipulation;
+using UnityEngine;
+
+public class AxisMovementConstraint : TransformConstraint
+{
+    #region Properties
+
+    [SerializeField] 
+    [Tooltip("Allow movement along an X-Axis")]
+    private bool movementOnXAxis;
+
+
+    [SerializeField]
+    [Tooltip("Allow movement along an Y-Axis")]
+    private bool movementOnYAxis;
+
+
+    [SerializeField]
+    [Tooltip("Allow movement along an Z-Axis")]
+    private bool movementOnZAxis;

```

Comment:
Why are movementOn*Axis serialized fields? You force this value to `true` in `Start`.   It looks like you are using this variable to track some private state, and it shouldn't be serialized.